### PR TITLE
rq list --all: Avoid conflict with the --state option

### PR DESCRIPTION
**Pull request by dmach**
Fixes: #1176

### CODE REVIEW

File: osc/commandline.py [hunk 1]
```diff
@@ -2300,6 +2300,12 @@ def do_request(self, subcmd, opts, *args):
             raise oscerr.WrongOptions('Sorry, the options \'--interactive\' and '
                                       '\'--non-interactive\' are mutually exclusive')
 
+        if opts.all:
+            state_list = ["all"]
+        else:
+            state_list = opts.state.split(",")
```
Comment (dirkmueller) [inline osc/commandline.py:2306]: where is the default handling gone to ?

File: osc/commandline.py [hunk 2]
```diff
@@ -2423,15 +2429,9 @@ def do_request(self, subcmd, opts, *args):
...
-                if state_list == ['']:
-                    state_list = ()
-                if opts.all:
-                    state_list = ['all']
-                else:
-                    for s in state_list:
-                        if s not in states and not s == 'all':
-                            raise oscerr.WrongArgs('Unknown state \'%s\', try one of %s' % (s, ','.join(states)))
+                for s in state_list:
+                    if s not in states and not s == 'all':
```
Comment (dirkmueller) [inline osc/commandline.py:2433]: NIT: I know it has been that way before

```suggestion
                    if s!= 'all' and s not in states:
```

File: osc/commandline.py (hunk 1 above)
Comment (dmach) [reply osc/commandline.py:2308]: Good catch, fixed.

File: osc/commandline.py (hunk 2 above)
Comment (dmach) [reply osc/commandline.py:2435]: The proposed change makes sense to me - it's definitely an improvement to code readability. Thanks.

**Review by dirkmueller — APPROVED** [review]

**Pull request merged**